library/std/src/sync/mod.rs

```diff
@@ -167,6 +167,11 @@
 
 #![stable(feature = "rust1", since = "1.0.0")]
 
+// No formatting: this file is just re-exports, and their order is worth preserving.
+#![cfg_attr(rustfmt, rustfmt::skip)]
```

Comment:
Could this skip just get applied to specific blocks, like in https://github.com/rust-lang/rust/blob/addbd001ec56741829f20a3000892f8620dd0843/library/core/src/unicode/mod.rs? Applying to the whole file would probably make it a bit too easy for things to get messy.

Reply:
Then pretty much every `use` block would have to be annotated, because otherwise `rustfmt` will regroup them.

IMO this looks even noisier and less accurate with all the additional attributes and inability to leave two blank lines between the groups (`rustfmt` removes double empty lines).

Reply:
Alright, the blank line section breaks are fine but could you remove the double blank lines since rustfmt doesn't do those anywhere? I.e. `\n\n\n` -> `\n\n`.